src/storage/src/hummock/file_cache/store.rs

```diff
@@ -12,431 +12,533 @@
-    pub fn is_empty(&self) -> bool {
-        self.len() == 0
+    fn read(mut buf: &[u8]) -> Self {
+        let sst_id = buf.get_u64();
+        let block_idx = buf.get_u64();
+        Self { sst_id, block_idx }
     }
+}
 
-    #[tracing::instrument(skip(self))]
-    pub async fn finish(mut self) -> Result<(Vec<K>, Vec<SlotId>)> {
-        // First free slots during last batch.
-
-        let mut freelist = Vec::with_capacity(FREELIST_DEFAULT_CAPACITY);
-        std::mem::swap(&mut *self.store.freelist.write(), &mut freelist);
-
-        if !freelist.is_empty() {
-            let mut guard = self
-                .store
-                .meta_file
-                .write()
-                .instrument(tracing::trace_span!("meta_write_lock_free_slots"))
-                .await;
-            for slot in freelist {
-                if let Some(bloc) = guard.free(slot) {
-                    let offset = bloc.bidx as u64 * self.block_size as u64;
-                    let len = bloc.blen(self.block_size as u32) as usize;
-                    self.store.cache_file.punch_hole(offset, len)?;
-                }
-            }
-            drop(guard);
-        }
-
-        if self.is_empty() {
-            return Ok((vec![], vec![]));
-        }
-
-        // Write new cache entries.
-
-        let mut slots = Vec::with_capacity(self.blocs.len());
-
-        let mut boff: Option<u32> = None;
-
-        for buffer in self.buffers {
-            let blen = buffer.len();
-
-            let timer = self.store.metrics.disk_write_latency.start_timer();
-            let offset = self.store.cache_file.append(buffer).await? / self.block_size as u64;
-            timer.observe_duration();
-
-            if boff.is_none() {
-                boff = Some(offset.try_into().unwrap());
-            }
-
-            self.store.metrics.disk_write_bytes.inc_by(blen as f64);
-            self.store.metrics.disk_write_io_size.observe(blen as f64);
-        }
-
-        let boff = boff.unwrap();
+impl Value for Box<Block> {
+    fn serialized_len(&self) -> usize {
+        self.raw_data().len()
+    }
 
-        for bloc in &mut self.blocs {
-            bloc.bidx += boff;
-        }
+    fn write(&self, mut buf: &mut [u8]) {
+        buf.put_slice(self.raw_data())
+    }
 
-        // Write guard is only needed when updating meta file, for data file is append-only.
-        let mut guard = self
-            .store
-            .meta_file
-            .write()
-            .instrument(tracing::trace_span!("meta_write_lock_update_slots"))
-            .await;
+    fn read(buf: &[u8]) -> Self {
+        let data = Bytes::copy_from_slice(buf);
+        let block = Block::decode_from_raw(data);
+        Box::new(block)
+    }
+}
 
-        for (key, bloc) in self.keys.iter().zip_eq_fast(self.blocs.iter()) {
-            slots.push(guard.insert(key, bloc)?);
-        }
+impl Value for Box<Sstable> {
+    fn weight(&self) -> usize {
+        std::mem::size_of::<Self>()
```

Comment:
Seems not accurate to use `sizeof` to get the memory usage if this method intended to calculate it.

Reply:
Oh, I forgot to modify this!

Reply:
Let me just remove it for now. `weight` is only used by foyer memory cache, which is not integrated yet.